tools/autograd/gen_python_functions.py

```diff
@@ -760,7 +760,8 @@ def sort_overloads(
 
     def is_arg_smaller(t1: Type, t2: Type) -> bool:
         return (str(t1) == 'Scalar' and str(t2) == 'Tensor' or
-                'Dimname' in str(t1) and 'Dimname' not in str(t2))
+                'Dimname' in str(t1) and 'Dimname' not in str(t2) or
+                str(t1) == 'int[]' and (str(t2) == 'int' or str(t2) == 'int?'))
```

Comment:
this deserves a little comment saying why

Reply:
I have added the comment, which also refers to the discussion in the given issue.